data/trie/pruningBuffer_test.go

```diff
@@ -0,0 +1,41 @@
+package trie
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func defaultPruningBuffer() *pruningBuffer {
+	sb := newPruningBuffer()
+	sb.add([]byte("0"))
+	sb.add([]byte("1"))
+
+	return sb
+}
+
+func TestPruningBuffer_NewPruningBuffer(t *testing.T) {
+	assert.NotNil(t, newPruningBuffer())
+}
+
+func TestSnapshotsBuffer_Add(t *testing.T) {
+	sb := defaultPruningBuffer()
+
+	assert.Equal(t, 2, len(sb.buffer))
+}
+
+func TestSnapshotsBuffer_Remove(t *testing.T) {
```

Comment:
add a test with remove at end of list. or having the same roothash multiple times.

Reply:
Done.